packages/pigeon/lib/java_generator.dart

```diff
@@ -606,46 +606,50 @@ class JavaGenerator extends StructuredGenerator<JavaOptions> {
                 : _javaTypeForDartType(method.returnType);
             indent.writeln(
                 'ArrayList<Object> wrapped = new ArrayList<Object>();');
-            indent.write('try ');
-            indent.addScoped('{', '}', () {
-              final List<String> methodArgument = <String>[];
-              if (method.arguments.isNotEmpty) {
+            final List<String> methodArgument = <String>[];
+            if (method.arguments.isNotEmpty) {
+              indent.writeln(
+                  'ArrayList<Object> args = (ArrayList<Object>) message;');
+              indent.write('if (args == null) ');
```

Comment:
We should not be doing runtime checks that Pigeon-generated messages follow the Pigeon protocol. To the extent that we have them in some languages in some places it's a legacy; we should be removing them rather than adding more.